### PR TITLE
Add typescript types FluidTimePicker and FluidTimePickerSelect

### LINKED ISSUE #17358: FluidTimePicker
Add TypeScript types to FluidTimePicker, FluidTimePickerSkeleton

### LINKED ISSUE #17359: FluidTimePickerSelect
Typescript types to FluidTimePickerSelect

### PR DESCRIPTION
**Pull request by riddhybansal**
Closes #17358 
Closes #17359 

Added Typescript types to FluidTimePicker and FluidTimePickerSelect and their skeleton state

#### Changelog

**New**

Added Typescript types to FluidTimePicker and FluidTimePickerSelect and there skeleton state
Converted index.js to tsx for both the component

#### Testing / Reviewing

FluidTimePicker and FluidTimePickerSelect and their skeleton state should work as intended with typescript support

### CODE REVIEW

**Comment by riddhybansal** [comment]
Hi there I have one doubt over here I ahve passes id as empty string as it is a required prop, Can we do anything else instead of here ?

**Comment by tay1orjones** [comment]
@riddhybansal I think FluidTimePickerProps needs to extend/inherit FluidTextInputProps

File: packages/react/src/components/FluidTimePicker/__tests__/FluidTimePicker-test.js [hunk 1]
```diff
@@ -15,28 +15,28 @@ const prefix = 'cds';
...
-          </FluidTimePickerSelect>
-          <FluidTimePickerSelect id="select-2" labelText="Timezone">
-            <SelectItem value="et" text="Eastern Time (ET)" />
-            <SelectItem value="ct" text="Central Time (CT)" />
-            <SelectItem value="mt" text="Mountain Time (MT)" />
-            <SelectItem value="pt" text="Pacific Time (PT)" />
-          </FluidTimePickerSelect>
-        </FluidTimePicker>
-      );
+    // it('should add extra classes that are passed via className', () => {
```
Comment (guidari) [inline packages/react/src/components/FluidTimePicker/__tests__/FluidTimePicker-test.js]: Hey @riddhybansal is this testing failing?

Comment (riddhybansal) [reply packages/react/src/components/FluidTimePicker/__tests__/FluidTimePicker-test.js]: oops I was testing something so commented that time need to uncomment it !!

**Review by guidari — APPROVED** [review]
LGTM!

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**